src/main/java/org/truffleruby/language/methods/LookupMethodNode.java

```diff
@@ -111,7 +115,17 @@ protected InternalMethod lookupMethodUncached(VirtualFrame frame, Object self, S
             throw new UnsupportedOperationException("method lookup not supported on foreign objects");
         }
 
-        final InternalMethod method = ModuleOperations.lookupMethodUncached(metaClass, name);
+        final InternalMethod method;
+        // Lookup first in the metaclass as we are likely to find the method there
+        final ModuleFields fields = Layouts.MODULE.getFields(metaClass);
+        InternalMethod topMethod;
+        if (noPrependedModulesProfile.profile(fields.getFirstModuleChain() == fields) &&
```

Comment:
There's quite a bit going on here. I was actively looking for the assignment to `topMethod` and missed it a couple times.

Reply:
Any idea how to simplify?
Duplicating the else branch doesn't seem so nice.
We could also just assign to `method` if it's not final.

Reply:
I see. I guess it's just a trade-off in complexity. Fortunately, I shouldn't have to encounter this terribly frequently (i.e., it's fine as-is) ;-)